neurodamus/node.py

```diff
@@ -1592,12 +1588,8 @@ def __init__(
                 1 - Info messages (default)
                 2 - Verbose
                 3 - Debug messages
-            cleanup_atexit: (bool) Call cleanup in the destructor
-                [for more see: https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-976]
```

Comment:
I think better to keep `cleanup_atexit` because I am using it for Polina's gap junction compensation
```
 nd = Neurodamus(config_file, auto_init=False, logging_level=log_level, cleanup_atexit=False, **options)
__ = load_user_modification(nd, node_pop="thalamus_neurons") # gj compensation
nd.init()
nd.run()
nd.cleanup()
```
I see that multiscale_run calls `cleanup_atexit=false` as well, https://bbpgitlab.epfl.ch/molsys/multiscale_run/-/blob/main/multiscale_run/neurodamus_manager.py?ref_type=heads#L20 . We don't know why we need to clean up later, as address in BBPBGLIB-976.

Reply:
`cleanup_atexit=false` is just a workaround introduced to avoid calling the destructor because it was creating problems. My change fixes exactly that case. We shouldn't ever avoid calling the destructor, the destructor should be safe. So now what we do is to call the cleanup procedure outside.

Reply:
ok, I understand better now.  Thank for the improvement. We can merge it together in this PR.  

BTW, then probably we can remove `cleanup_atexit=True` from the `Neurodamus` class constructor ? Also ping multiscale_run ppl to change their script.

Reply:
Oh, yes, it was an oversight to leave it there